tests/client/handlers/vaults.test.ts

```diff
@@ -1341,18 +1342,149 @@ describe('vaultsSecretsMkdir', () => {
     const vaultName = 'test-vault';
     const vaultId = await vaultManager.createVault(vaultName);
     const dirPath = 'dir/dir1/dir2';
-    const response = await rpcClient.methods.vaultsSecretsMkdir({
-      recursive: true,
+    const response = await rpcClient.methods.vaultsSecretsMkdir();
+    const writer = response.writable.getWriter();
+    await writer.write({
       nameOrId: vaultsUtils.encodeVaultId(vaultId),
       dirName: dirPath,
+      metadata: { options: { recursive: true } },
     });
-    expect(response.success).toBeTruthy();
+    await writer.close();
+
+    for await (const data of response.readable) {
+      expect(data.type).toEqual('success');
+    }
     await vaultManager.withVaults([vaultId], async (vault) => {
       await vault.readF(async (efs) => {
         expect(await efs.exists(dirPath)).toBeTruthy();
       });
     });
   });
+  test('fails to make directories without recursive', async () => {
+    const vaultName = 'test-vault';
+    const vaultId = await vaultManager.createVault(vaultName);
+    const encodeVaultId = vaultsUtils.encodeVaultId(vaultId);
+    const dirPath = 'dir/dir1/dir2';
+    const response = await rpcClient.methods.vaultsSecretsMkdir();
+    const writer = response.writable.getWriter();
+    await writer.write({ nameOrId: encodeVaultId, dirName: dirPath });
+    await writer.close();
+    for await (const data of response.readable) {
+      expect(data.type).toEqual('error');
+      if (data.type !== 'error') utils.never();
+      expect(data.code).toEqual('ENOENT');
+      expect(data.reason).toEqual(dirPath);
+    }
```

Comment:
Apparently the LSP complaints of it not being able to properly narrow down the type wasn't an issue in my LSP, but rather, the code.

https://github.com/typescript-language-server/typescript-language-server/issues/924

I raised this issue on my LSP's issue page, and there, I was informed that `ts-node` doesn't actually use typescript, and if I ran `npx tsc` then it should give me the same errors, which was the case.

As such, while this works on `ts-node`, if we ever switched compilers, this would just fail to build. In my opinion, this needs to be resolved rather than ignoring this as it isn't cropping up for now.

Reply:
After talking about this a bit more with Brian, we have realised that the types are correct, but typescript cannot evaluate this type properly as it is too complicated. As such, while not ideal, but this is one time that I will be allowed to use the `as` keyword to explicitly change the type of an object.

Reply:
The problem is the utility type in `ClientRPCResponseResult` that's wrapping the `SuccessOrErrorMessage`. Its preventing typescripts ability to narrow the type down. We've had issues with utility types and  type inference like this before.